src/client/cached.go

```diff
@@ -72,7 +82,24 @@ func (d *DiscoveryCacher) discoverAndCache(timeout time.Duration) (HTTPClient, e
 	if err != nil {
 		d.Logger.WithError(err).Warnf("while storing %q in the cache", d.StorageKey)
 	}
-	return client, nil
+	return d.wrap(client, timeout), nil
+}
+
+func Expired(currentTime time.Time, creationTimestamp int64, ttl time.Duration, jitterMaxPercentage uint) bool {
+	rand.Seed(time.Now().UTC().UnixNano())
+
+	selectedJitter := rand.Intn(int(jitterMaxPercentage)+1) - 1
+
+	if i := rand.Intn(2); i == 1 {
+		selectedJitter = selectedJitter * -1.0
+	}
+
+	jitterPercentage := (100 + float64(selectedJitter)) / 100
```

Comment:
What about getting `rand.Float64`, substract `0.5` from it, and then multiply that by `float64(jitterMaxPercentage)/100` ?

Reply:
I decided to use `rand.Intn` to avoid using floating point as long as possible, which I think makes debugging simpler.

I'll actually fix `-1.0` to `-1`.

Reply:
So `jitterPercentage := (100 + float64(selectedJitter)) / 100` produces a float as it does the division, then next line converts it back to time.Duration.

Reply:
I would be more concerned about readability than debuggability. Are you sure it's better to have 4 lines of arithmetic an and if, as opposed to two?

Reply:
>What about getting (rand.Float64() * 2) - 1, which would be a random number between -1 and 1, and then multiply that by float64(jitterMaxPercentage)/100 ?

With that, we still have to do an operation on the TTL, either addition or multiplication by factor.

Given that `(rand.Float64() * 2) - 1` gives us let's say `-0.5`.

Then `(jitterMaxPercentage/100)*-0.5` gives `−0.1`, with `jitterMaxPercentage = 20`.

How can we apply it to original TTL then?

Reply:
Okay, it can be done this way:
```go
  // Convert e.g. 20% to 0.2
  jitterPercentage := float64(jitterMaxPercentage) / 100

  // Random number between -1 and 1.
  randomFactor := ((rand.Float64() * 2) - 1)

  // TTL multiplier must be a positive value for jitterMaxPercentage < 100, so add 1 to ensure that.
  //
  // With jitterMaxPercentage < 100:
  //
  // 0.2 * -0.75 = -0.15
  // -0.15 + 1 = 0.85
  //
  // With jitterMaxPercentage > 100:
  //
  // 1.2 * -1.5 = -1.8
  // -1.8 + 1 = -0.8
  //
  ttlMultiplier := (jitterPercentage * randomFactor) + 1

  // Multiply TTL as float with multiplier, then convert back to duration in seconds.
  ttlWithJitter := time.Duration(ttl.Seconds()*ttlMultiplier) * time.Second

  // As in documentation, time.Now().Sub() is the same as time.Since().
  return currentTime.Sub(time.Unix(creationTimestamp, 0)) > ttlWithJitter
```